core/src/Lender.sol

```diff
@@ -133,27 +135,29 @@ contract Lender is Ledger {
     }
 
     function redeem(uint256 shares, address recipient, address owner) public returns (uint256 amount) {
-        // Guard against reentrancy, accrue interest, and update reserves
-        (Cache memory cache, uint256 inventory) = _load();
-
-        amount = _convertToAssets(shares, inventory, cache.totalSupply, /* roundUp: */ false);
-        require(amount != 0, "Aloe: zero impact");
-
         if (msg.sender != owner) {
             uint256 allowed = allowance[owner][msg.sender];
             if (allowed != type(uint256).max) allowance[owner][msg.sender] = allowed - shares;
         }
```

Comment:
This is a "check" that we can perform without loading or computing anything else, so it goes first.